crates/wasi-http/src/types.rs

```diff
@@ -257,6 +257,13 @@ pub struct HostOutgoingRequest {
     pub body: Option<HyperOutgoingBody>,
 }
 
+#[derive(Default)]
+pub struct HostRequestOptions {
+    pub connect_timeout: Option<u32>,
```

Comment:
Lets use either `std` or `tokio::time::Duration` to represent these values in the host as early as we can.

Reply:
The downside is that the getters become quite fallible as they're turning a `std::time::Duration` back into a `u32`.

Reply:
Isnt that resolved when we switch to using a monotonic duration, which is a u64?

Reply:
`std::time::Duration::as_millis` returns a u128, so we won't avoid it completely.

Reply:
I implemented this by treating conversion failure from `u128` as an `Ok(None)` return value. Since we default all these timeouts to `None` anyway, I think a conversion failure would mean that the guest somehow managed to set a larger than u32 value to begin with, then tried to get one back. The host setting an invalid default would be the other possibility, but we don't do this anywhere in wasmtime currently.